hat/app/org/hatdex/hat/utils/NamespaceUtils.scala

```diff
@@ -46,4 +47,55 @@ object NamespaceUtils {
     }
     matchedRoles.flatten.nonEmpty
   }
+
+  def getWriteNamespace(roles: Seq[UserRole]): Option[String] = {
+    val writeNamespaces = roles.map {
+      case NamespaceWrite(n) => n
+      case _                 => ""
+    }
+
+    if (writeNamespaces.isEmpty)
+      None
+    else
+      Some(writeNamespaces.head)
+  }
+
+  def getReadNamespace(roles: Seq[UserRole]): Option[String] = {
+    val readNamespaces = roles.map {
+      case NamespaceRead(n) => n
+      case _                => ""
+    }
+
+    if (readNamespaces.isEmpty)
+      None
+    else
+      Some(readNamespaces.head)
+  }
+
+  def verifyNamespace(
+      app: Application,
+      namespace: String): Boolean = {
+
+    val canReadNamespace  = verifyNamespaceRead(app, namespace)
+    val canWriteNamespace = verifyNamespaceWrite(app, namespace)
+
+    (canReadNamespace || canWriteNamespace)
+  }
+
+  def verifyNamespaceRead(
+      app: Application,
+      namespace: String): Boolean = {
+    val rolesOk = testReadNamespacePermissions(app.permissions.rolesGranted, namespace)
```

Comment:
there is no point creating a variable to discard it immediately, this is just syntactic noise in my opinion
I would also argue we no not need this method at all, given the logic is already in another existing method, I would just rename the existing method (`testReadNamespacePermissions` becomes `canReadNamespace`)